core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/MetadataSetupService.java

```diff
@@ -361,6 +367,29 @@ public void saveLoadBalancerMetadata(Stack stack, Iterable<CloudLoadBalancerMeta
                 loadBalancerEntry.setIp(cloudLoadBalancerMetadata.getIp());
                 loadBalancerEntry.setType(cloudLoadBalancerMetadata.getType());
                 String endpoint = loadBalancerConfigService.generateLoadBalancerEndpoint(stack);
+
+                List<StackStatusView> stoppedDatalakes = stackService.getByEnvironmentCrnAndStackType(stack.getEnvironmentCrn(), StackType.DATALAKE)
+                        .stream().filter(s -> s.getStatus().getStatus().isStopState()).collect(Collectors.toList());
+                if (!stoppedDatalakes.isEmpty()) {
+                    /* Starts to check for a situation where we are resizing a datalake that did not previously have loadbalancers
+                        so that we can use the same endpoint name for a seamless transistion
+                     */
+                    LOGGER.info("Using old datalake endpoint name for resized datalake: {}, env: {}", stack.getName(), stack.getEnvironmentCrn());
+                    if (stoppedDatalakes.size() > 1) {
+                        String ids = stoppedDatalakes.stream().map(StackStatusView::getId).map(Object::toString).collect(Collectors.joining(","));
+                        LOGGER.warn("more than one datalake found to resize from: {}", ids);
+                    }
+                    Long oldId = stoppedDatalakes.get(0).getId();
+
+                    Set<LoadBalancer> oldLoadbalancers = loadBalancerPersistenceService.findByStackId(oldId);
+                    if (oldLoadbalancers.isEmpty()) {
+                        Stack oldStack = stackService.getByIdWithGatewayInTransaction(oldId);
+                        if (stack.getDisplayName().equals(oldStack.getDisplayName())) {
+                            endpoint = oldStack.getPrimaryGatewayInstance().getShortHostname();
+                        }
+                    }
```

Comment:
What happens in the case that `!oldLoadbalancers.isEmpty()`? I'm assuming that the new DL has the same name as the old and so the LB endpoint between the two will be identical. Is that accurate?

Reply:
Yes, It's already the case that the new one will use the correct name, I've added this to be specific to just the datalakes that were created without a load balancer